src/cloudwatch-mcp-server/awslabs/cloudwatch_mcp_server/cloudwatch_metrics/tools.py

```diff
@@ -792,30 +881,242 @@ def _parse_alarm_recommendation(self, alarm_data: Dict[str, Any]) -> AlarmRecomm
+
+    def _parse_metric_dimensions(self, alarm_data: Dict[str, Any]) -> List[str]:
+        """Parse metric dimensions from the alarm data.
+
+        Args:
+            alarm_data: Raw alarm recommendation data
+
+        Returns:
+            AlarmRecommendation: Parsed alarm recommendation object
+        """
+        dimensions = []
+        for dim_data in alarm_data.get('dimensions', []):
+            alarm_dim = AlarmRecommendationDimension(
+                name=dim_data.get('name', ''),
+                value=dim_data.get('value') if 'value' in dim_data else None,
+            )
+            dimensions.append(alarm_dim)
+
+        return dimensions
+
+    def _calculate_period_seconds(self, analysis_period_minutes: int) -> int:
+        """Calculate period in seconds based on analysis period."""
+        # Use the same logic as _prepare_time_parameters
+        time_window_seconds = analysis_period_minutes * 60
+        return max(60, time_window_seconds // analysis_period_minutes)
+
+    def _parse_metric_data_response(
+        self, response: GetMetricDataResponse, period_seconds: int
+    ) -> MetricData:
+        """Parse CloudWatch GetMetricData response into MetricData."""
+        timestamps = []
+        values = []
+
+        if response.metricDataResults and response.metricDataResults[0].datapoints:
+            datapoints = response.metricDataResults[0].datapoints
+            timestamps_ms = [int(dp.timestamp.timestamp() * 1000) for dp in datapoints]
+            raw_values = [dp.value for dp in datapoints]
+
+            sorted_data = sorted(zip(timestamps_ms, raw_values))
+            if sorted_data:
+                timestamps, values = zip(*sorted_data)
+                timestamps = list(timestamps)
+                values = list(values)
+
+        return MetricData(period_seconds=period_seconds, timestamps=timestamps, values=values)
+
+    async def analyze_metric(
+        self,
+        ctx: Context,
+        namespace: str = Field(
+            ..., description="The namespace of the metric (e.g., 'AWS/EC2', 'AWS/Lambda')"
+        ),
+        metric_name: str = Field(
+            ..., description="The name of the metric (e.g., 'CPUUtilization', 'Duration')"
+        ),
+        dimensions: List[Dimension] = Field(
+            default_factory=list,
+            description='List of dimensions that identify the metric, each with name and value',
+        ),
+        region: Annotated[
+            str,
+            Field(description='AWS region to query. Defaults to us-east-1.'),
+        ] = 'us-east-1',
+        analysis_period_minutes: Annotated[
+            int,
+            Field(
+                description='Number of minutes of historical data to analyze. Defaults to 2 weeks (20160 minutes).'
+            ),
+        ] = DEFAULT_ANALYSIS_PERIOD,
+    ) -> Dict[str, Any]:
+        """Analyzes CloudWatch metric data to determine seasonality, trend, data density and statistical properties.
```

Comment:
We should be frugal about tool descriptions, as they gets loaded in the context on the consumers. Let's try to be very short and crisp here (see descriptions from other tools).